Src/FluentAssertions/Xml/XAttributeAssertions.cs

```diff
@@ -30,12 +32,23 @@ public XAttributeAssertions(XAttribute attribute)
         /// <param name="becauseArgs">
         /// Zero or more objects to format using the placeholders in <paramref name="because" />.
         /// </param>
+        /// <exception cref="ArgumentNullException"><paramref name="expected"/> is <c>null</c>.</exception>
         public AndConstraint<XAttributeAssertions> Be(XAttribute expected, string because = "", params object[] becauseArgs)
         {
-            Execute.Assertion
-                .ForCondition(Subject.Name == expected.Name && Subject.Value == expected.Value)
+            Guard.ThrowIfArgumentIsNull(expected, nameof(expected));
```

Comment:
🤔 I wonder about this one. What if you're comparing two references to an `XAttribute` and both are `null`. Shouldn't this assertion succeed? This is also how `StringAssertions` work.

Reply:
Yeah, that might be a good exception.

Reply:
And the same applies for `[Not]BeEquivalentTo`

Reply:
It was more work then expected.
Also added more tests to cover these scenarios.